src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/EvidenceUploadApplicantNotificationHandlerTest.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.civil.handler.callback.camunda.notification;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.config.properties.notification.NotificationsProperties;
+import uk.gov.hmcts.reform.civil.handler.callback.BaseCallbackHandlerTest;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.sampledata.CallbackParamsBuilder;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
+import uk.gov.hmcts.reform.civil.service.NotificationService;
+
+import java.util.Map;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.handler.callback.camunda.notification.NotificationData.CLAIM_REFERENCE_NUMBER;
+
+@SpringBootTest(classes = {
+        EvidenceUploadApplicantNotificationHandler.class,
+        JacksonAutoConfiguration.class
+})
+class EvidenceUploadApplicantNotificationHandlerTest extends BaseCallbackHandlerTest {
+
+    @MockBean
+    private NotificationService notificationService;
+    @MockBean
+    private NotificationsProperties notificationsProperties;
+    @Autowired
+    private EvidenceUploadApplicantNotificationHandler handler;
+
+    @Nested
+    class AboutToSubmitCallback {
+
+        @BeforeEach
+        void setup() {
+            when(notificationsProperties.getEvidenceUploadTemplate()).thenReturn("template-id");
+        }
+
+        @Test
+        void shouldNotifyApplicantSolicitor_whenInvoked() {
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimDetailsNotified().build();
+            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_SUBMIT, caseData).build();
+
+            handler.handle(params);
+
+            verify(notificationService).sendMail(
+                    "applicantsolicitor@example.com",
+                    "template-id",
+                    getNotificationDataMap(caseData),
+                    "evidence-upload-notification-000DC001"
+            );
+        }
```

Comment:
Let's start putting `//Given`, `//When`, `//Then` comments in our tests. It's trivial, but it does increase clarity in a context where we can't do proper BDD tests.